src/deep-kernel/lib/Config/Driver/AsyncConfig.js

```diff
@@ -0,0 +1,99 @@
+/**
+ * Created by AlexanderC on 3/7/16.
+ */
+
+'use strict';
+
+import {AbstractDriver} from './AbstractDriver';
+import {HttpDriver} from './HttpDriver';
+import AWS from 'aws-sdk';
+
+export class AsyncConfig extends AbstractDriver {
+  /**
+   * @param {String} bucket
+   * @param {String} configFile
+   */
+  constructor(bucket = null, configFile = AsyncConfig.DEFAULT_CONFIG_FILE) {
+    super();
+
+    this._bucket = bucket;
+    this._configFile = configFile;
+  }
+
+  /**
+   * @param {Kernel|*} kernel
+   */
+  static createFromKernel(kernel) {
+    return new AsyncConfig(kernel.isBackend ? kernel.get('fs').public.bucket : null);
+  }
+
+  /**
+   * @returns {String|*}
+   */
+  get bucket() {
+    return this._bucket;
+  }
+
+  /**
+   * @returns {String|*}
+   */
+  get configFile() {
+    return this._configFile;
+  }
+
+  /**
+   * @param {String} bucket
+   * @param {String} configFile
+   * @private
+   */
+  _load(bucket = null, configFile = null) {
+    this._bucket = bucket || this._bucket;
+    this._configFile = configFile || this._configFile;
+
+    this._bucket ? this._loadFromS3() : this._loadFromEndpoint();
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Expected an assignment or function call and instead saw an expression.](https://www.codacy.com/app/MitocGroup/deep-framework/file/2202947102/issues/source?bid=2895475&fileBranchId=3169716#l53)